File: packages/sdk/java/src/main/java/com/ihui/ai/sdk/ServerException.java
package com.ihui.ai.sdk;

/**
 * 5xx 服务端错误异常。
 */
public class ServerException extends SdkException {

    private static final long serialVersionUID = 1L;

    /**
     * 构造 ServerException。
     *
     * @param status  HTTP 状态码(应为 5xx)
     * @param code    错误码
     * @param message 错误消息
     * @param details 错误详情
     */
    public ServerException(int status, String code, String message, Object details) {
        super(status, code, message, details);
    }
}

File: packages/sdk/dotnet/Client/ServerException.cs
namespace Ihui.AI;

/// <summary>
/// 5xx 服务端错误异常。
/// </summary>
public class ServerException : SdkException
{
    /// <summary>构造 ServerException。</summary>
    /// <param name="status">HTTP 状态码(5xx)</param>
    /// <param name="code">错误码</param>
    /// <param name="message">错误消息</param>
    /// <param name="details">错误详情</param>
    public ServerException(int status, string? code, string message, object? details = null)
        : base(status, code, message, details)
    {
    }
}
